app/views/casa_org/_contact_topics.html.erb

```diff
@@ -0,0 +1,69 @@
+<div class="row">
+  <div class="col-lg-12">
+    <div class="card-style mb-30">
+      <div class="row align-items-center">
+        <div class="col-md-6">
+          <h3>Contact Topics</h3>
+        </div>
+        <div class="col-md-6">
+          <div class="breadcrumb-wrapper">
+            <span class="ml-5">
+              <%= link_to new_contact_topic_path, class: "btn-sm main-btn primary-btn btn-hover" do %>
+                <i class="lni lni-plus mr-10"></i>
+                New Contact Topic
+              <% end %>
+            </span>
+          </div>
+        </div>
+      </div>
+      <div class="table-wrapper table-responsive">
+        <table class="table striped-table" id="contact-topics">
+          <thead>
+            <tr>
+              <th>Question</th>
+              <th>Details</th>
+              <th>Active?</th>
+            </tr>
+          </thead>
+          <tbody>
+            <% @contact_topics.each do |contact_topic| %>
+              <tr id="contact_topic-<%= contact_topic.id %>">
+                <td scope="row" class="min-width">
+                  <%= contact_topic.question %>
+                </td>
+                <td scope="row" class="min-width pre-line"><%= contact_topic.details %></td>
+                <td scope="row" class="min-width">
+                  <%= contact_topic.active ? "Yes" : "No" %>
+                </td>
+                <td>
+                  <%= render(KebabMenuComponent.new(menu_title: "Actions Menu")) do %>
+                    <li><%= link_to "Edit", edit_contact_topic_path(contact_topic), class: "dropdown-item" %></li>
+                    <li><%= render(ModalComponent.new( id: "contact_topic-#{contact_topic.id}", modal: false, button_value: "Delete")) %></li>
+                  <% end %>
+                </td>
+              </tr>
+              <%= render(ModalComponent.new(
+                id: "contact_topic-#{contact_topic.id}",
+                button: false,
+                header_text: "Delete Court Report Topic?"
+              )) do |component| %>
+                <% component.with_modal_footer do %>
+                  <%= link_to soft_delete_contact_topic_path(contact_topic), method: :delete,
+                    class: "btn-sm main-btn danger-btn btn-hover ms-auto" do %>
+                    <i class="lni lni-trash-can mr-10"></i>
+                    Delete Court Report Topic
+                  <% end %>
+                <% end %>
+                <% component.with_modal_content do %>
+                  <p>This topic and its related questions will be deleted and will no longer be presented while filling out case contacts.</p>
+                  <br>
+                  <p>This will not affect case contacts that have already been created.</p>
```

Comment:
How about building this more into the ModalComponent itself. Create a dir for Modals:
```
# not including the html templates here, but they should all have one
app/component/modal/
  - modal_component.rb
  - content_component.rb
  - footer_component.rb
  - header_component.rb
```

Reply:
Sidebar is a smaller example for this.

Reply:
How about?
app/component/modal/
  - group_component.rb
  - content_component.rb
  - footer_component.rb
  - header_component.rb
  - button_component.rb
  - link_component.rb

I really want to include a way to generate a link/button associated with a modal. Cause that button/link need some bootstrap data attrs that I would like to abstract away. Also in some situations a <a> might make more sense vs a <button> so providing both would be nice. (This is actually one of those situation, the stuff in the dropdown renders different cause 1 is a button and the other is a link)

Reply:
Sounds good